crates/forge/src/running.rs

```diff
@@ -68,17 +69,39 @@ fn build_hints_dict<'b>(
     (hints_dict, string_to_hint)
 }
 
-#[allow(clippy::too_many_arguments)]
+pub(crate) async fn blocking_run_from_test(
+    package_root: Arc<Utf8PathBuf>,
+    runner: Arc<SierraCasmRunner>,
+    case: Arc<TestCase>,
+    runner_config: Arc<RunnerConfig>,
+    runner_params: Arc<RunnerParams>,
+    args: Vec<Felt252>,
+) -> Result<TestCaseSummary> {
+    tokio::task::spawn_blocking(move || {
+        run_from_test_case(
+            &package_root,
+            &runner,
+            &case,
+            &runner_config,
+            &runner_params,
+            args,
+        )
+    })
+    .await?
+}
```

Comment:
We have a async threads (created in `lib.rs`) spawning blocking threads inside them. This doesn't seem like a correct solution. Why do we need this?

Reply:
Because cast block thread to call externals